engine/include/cubos/engine/render/shadows/directional_caster.hpp

```diff
@@ -0,0 +1,28 @@
+/// @file
+/// @brief Component @ref cubos::engine::DirectionalShadowCaster.
+/// @ingroup render-shadows-plugin
+
+#pragma once
+
+#include <vector>
+
+#include <cubos/core/reflection/reflect.hpp>
+
+#include <cubos/engine/api.hpp>
+#include <cubos/engine/render/shadows/caster.hpp>
+
+namespace cubos::engine
+{
+    /// @brief Component which enables shadow casting on a directional light.
+    /// @ingroup render-shadows-plugin
+    struct CUBOS_ENGINE_API DirectionalShadowCaster
+    {
+        CUBOS_REFLECT;
+
+        cubos::engine::ShadowCaster options;
```

Comment:
Hm I think we could choose a better field name, as the other fields are also options.
Maybe `base` or `common`?

Reply:
I agree with this, `base` or `baseSettings` makes the most sense to me

Reply:
`baseSettings` sounds the most clear to me.